cohortextractor/cohortextractor.py

```diff
@@ -206,6 +210,8 @@ def _generate_cohort(
             if index_date is not None:
                 logger.info(f"Setting index_date to {index_date}")
                 study.set_index_date(index_date)
+                if sql_logged:
+                    study.set_sql_logging(truncate=True)
```

Comment:
What is the scope of `study` here? 

It it possible the same `study` object will be later reused for non-index date queries, and thus truncate them when we didn't intend to?

Reply:
No, it shouldn't.  It gets instantiated in this function, before the index dates loop, with `load_study_definition`, at which point it has the default truncate=False, and the function doesn't return anything after the loop